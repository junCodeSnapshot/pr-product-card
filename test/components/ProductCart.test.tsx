import React from "react";
import ReactTestRenderer from "react-test-renderer";
import { ProductCart } from "../../src/components";
import { product1 } from "../data/productData";

const { act } = ReactTestRenderer;

describe('Test on Product Cart', () => {

    test('Should load propertly', () => {
        const wrapper = ReactTestRenderer.create(
            <ProductCart product={product1}>
                {
                    () => (
                        <h1>Product Card</h1>
                    )
                }
            </ProductCart>
        )
        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    test('Should increase the counter', () => {
        const wrapper = ReactTestRenderer.create(
            <ProductCart product={product1}>
                {
                    ({count, increaseBy}) => (
                        <div>
                            <h1>{count}</h1>
                            <button onClick={() => increaseBy(1)}>+1</button>
                        </div>
                    )
                }
            </ProductCart>
        );
        expect(wrapper.toJSON()).toMatchSnapshot();
        act(() => {
            wrapper.root.findByType('button').props.onClick();
        });
        expect((wrapper as any).toJSON().children[0].children[0].children[0]).toBe("1");
    })


})