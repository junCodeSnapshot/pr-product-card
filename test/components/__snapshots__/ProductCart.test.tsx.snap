// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on Product Cart Should increase the counter 1`] = `
<div
  className=" productCard undefined"
>
  <div>
    <h1>
      0
    </h1>
    <button
      onClick={[Function]}
    >
      +1
    </button>
  </div>
</div>
`;

exports[`Test on Product Cart Should load propertly 1`] = `
<div
  className=" productCard undefined"
>
  <h1>
    Product Card
  </h1>
</div>
`;
